### PR TITLE
Support Elastic primitive for UUID

### PR DESCRIPTION
**Pull request by markaya**
Summary: 
- Introduce Elastic Primitive for java.UUID
- Add tests for creating query with each Elastic Primitive

### CODE REVIEW

File: modules/library/src/test/scala/zio/elasticsearch/ElasticPrimitivesSpec.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      },
+      test("successfully create matches query with Double") {
+        assert(matches(FieldName, 1.00))(equalTo(Match[Any, Double](FieldName, 1.00, boost = None)))
+      },
+      test("successfully create matches query with Int") {
+        assert(matches(FieldName, 1))(equalTo(Match[Any, Int](FieldName, 1, boost = None)))
+      },
+      test("successfully create matches query with Long") {
+        assert(matches(FieldName, 1.toLong))(equalTo(Match[Any, Long](FieldName, 1.toLong, boost = None)))
+      },
```
Comment (dbulaja98) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticPrimitivesSpec.scala:35]: Can you do something like `1L`?

File: modules/library/src/test/scala/zio/elasticsearch/ElasticPrimitivesSpec.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import zio.elasticsearch.ElasticQuery._
+import zio.elasticsearch.query.Match
+import zio.test.Assertion.equalTo
+import zio.test._
+
+import java.util.UUID
+
+object ElasticPrimitivesSpec extends ZIOSpecDefault {
+
+  private val FieldName = "field_name"
```
Comment (dbulaja98) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticPrimitivesSpec.scala:12]: Rename this to `fieldName`.

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: modules/library/src/test/scala/zio/elasticsearch/ElasticPrimitivesSpec.scala (hunk 2 above)
Comment (dbulaja98) [reply modules/library/src/test/scala/zio/elasticsearch/ElasticPrimitivesSpec.scala:12]: Or make it final.

File: modules/library/src/test/scala/zio/elasticsearch/ElasticPrimitivesSpec.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package zio.elasticsearch
+
+import zio.elasticsearch.ElasticQuery._
+import zio.elasticsearch.query.Match
+import zio.test.Assertion.equalTo
+import zio.test._
+
+import java.util.UUID
+
+object ElasticPrimitivesSpec extends ZIOSpecDefault {
```
Comment (drmarjanovic) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticPrimitivesSpec.scala:10]:
```suggestion
object ElasticPrimitiveSpec extends ZIOSpecDefault {
```

**Review by drmarjanovic — APPROVED** [review]

**Pull request merged**